### PR TITLE
[Batch mode] WIP: Moving main outputs.

### PR DESCRIPTION
**Pull request by davidungar**
For batch mode, the output file names and overriding buffers need to be associated with the inputs, since each (primary) input can have its own set of outputs. This PR takes the first step of moving the main outputs into the inputs (avoiding a file-name-based lookup). It includes a rename from FrontendInputs to FrontendInputsAndOutputs. Once this PR lands, we can start moving over the supplementary outputs as well.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: include/swift/Frontend/FrontendInputsAndOutputs.h [hunk 1]
```diff
@@ -79,6 +84,9 @@ class FrontendInputsAndOutputs {
 
   bool isWholeModule() const { return !hasPrimaryInputs(); }
 
+  // FIXME: dmu fix uses / remove these when batch mode works
```
Comment (jrose-apple) [inline include/swift/Frontend/FrontendInputsAndOutputs.h]: Nitpick: please don't leave the personal FIXMEs in the code. If you want to leave a more long-lasting, complete-sentence one, that's fine.

File: include/swift/Frontend/InputFile.h [hunk 2]
```diff
@@ -41,7 +41,9 @@ class InputFile {
   /// string.
   InputFile(StringRef name, bool isPrimary,
             llvm::MemoryBuffer *buffer = nullptr)
-      : Filename(name), IsPrimary(isPrimary), Buffer(buffer) {
+      : Filename(
+            convertBufferNameFromLLVM_getFileOrSTDIN_toSwiftConventions(name)),
```
Comment (jrose-apple) [inline include/swift/Frontend/InputFile.h]: Hm, I actually *don't* like this change. It seems like it's not InputFile's job to know that the conventions are different.

**Review by jrose-apple — COMMENTED** [review]
I haven't looked at the last commit yet, and I basically skipped the first one because it's just a lot of renaming and reformatting. A few comments during your WIP.

**Pull request closed without merging**